### ISSUE TITLE
Overriding preset rules work on CI, but not on my local machine

### ISSUE THREAD

**Issue by jeddy3** [wooorm/remark-lint#96]
Hello @wooorm.

In https://github.com/stylelint/stylelint/pull/1844 I'm proposing we use the `consistent` and `recommended` presets (with, perhaps, an override for the `list-item-indent` rule). That PR is green-lit by both `travis` and `appveyor`. However, on my local machine I get 1038 errors like so:

```shell
13:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
```

All the errors are for the `list-item-indent` rule. It seems like the value for that rule isn't being changed from `"tab-size"` to `"space"` by the following config:

```json
{  
  "presets": [
    "remark-preset-lint-recommended",
    "remark-preset-lint-consistent"
  ],
  "plugins": {
    "lint": {
      "list-item-indent": "space"
    }
  }
}
```

Both CI's and my local machine appear to have the same node and npm versions:

```console
Richards-MacBook-Pro:stylelint jeddy3$ node --version
v6.3.0
Richards-MacBook-Pro:stylelint jeddy3$ npm --version
3.10.3
Richards-MacBook-Pro:stylelint jeddy3$ npm ls --depth=0
stylelint@7.1.0 /Users/jeddy3/Projects/stylelint
├── remark-cli@2.0.0
├── remark-lint@5.0.1
├── remark-preset-lint-consistent@1.0.0
├── remark-preset-lint-recommended@1.0.0
```

I'm on OSX, whereas the CIs are linux and windows.

Do you have any ideas what it might be? I figured I'd ask first in case anything springs to your mind. If not, I'll dig deeper myself.

**Comment by wooorm** [comment]
Thanks for opening, @jeddy3!

Presets are in essence config files published on npm. Meaning, they have their own dependencies as well. So, both presets have `remark-lint` installed too. Thus, [L84](https://github.com/stylelint/stylelint/pull/1844/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R84) can be dropped.

I’m guessing that you have an older version of remark-lint somewhere. `rm -rf node_modules/; npm install` should probably do the trick.

If that doesn’t work, you can add the `DEBUG` env variable when running `lint:md`, like so: `DEBUG="*" npm run lint:md`, which should specify where `remark-lint` is loaded from. If the previous didn’t work, could you show me the output of lint:md with `DEBUG`?

**Comment by jeddy3** [comment]
@wooorm Thanks for getting back to me :)

> So, both presets have remark-lint installed too. Thus, L84 can be dropped.

Good to know. I've changed our `package.json` accordingly.

> I’m guessing that you have an older version of remark-lint somewhere

That was my first thought too, and so I removed the `node_modules` directory before opening the issue. I just removed it again to be sure.

> If the previous didn’t work, could you show me the output of lint:md with DEBUG?

Is this the useful bit?:

https://gist.github.com/jeddy3/44bfe75446501171e59633ccfa8a2277

---

Having read https://github.com/stylelint/stylelint/pull/1844#issuecomment-242903750, I suspect we'll use the presets outright and so this issue is going to go away. If nothing stands out in the debug logs feel free to close issue. It's likely something amiss with my machine.

**Comment by wooorm** [comment]
I’m guessing the “culprit” here is `~/Projects/.remarkrc`. Could you check, if you rename it to `.remarkrc.bak` or something, whether this problem goes away?

My hypothesis is that:

* `~/Projects/.remarkrc` defines `list-item-indent` as `false` (`{"list-item-indent": false}`);
* presets define `listItemIndent` as `'tab-size'` (`{"list-item-indent": false, "listItemIndent": "tab-size"}`);
* `remarkConfig` in `package.json` overwrites `list-item-indent` (`{"list-item-indent": "space", "listItemIndent": "tab-size"}`);
* Object is “decamelised” in key order, where the last one wins, resulting in `{"list-item-indent": "tab-size"}`.

If you can go without `~/Projects/.remarkrc`, the easiest would be to remove that (for now).
I’ve opened wooorm/unified-engine/issues/14 for a proper fix, but that will take a while to land!

**Comment by jeddy3** [comment]
> If you can go without ~/Projects/.remarkrc, the easiest would be to remove that (for now).
